backend/prometheus.go

```diff
@@ -54,6 +54,13 @@ func (backend *Config) Collect(ch chan<- prometheus.Metric) {
 			backend.PrometheusSend(ch, point)
 		case <-*channels.Done:
 			recdone = true
+			if !rectimer.Stop() {
```

Comment:
Added this block because:
If all points are sent, and then the timer expires, then its stopped.
Then recdone is set to true, but the timer never triggers again, so you end up in an endless loop!